### PR TITLE
Developer branch

### PR DESCRIPTION
**Pull request by TimaVysotski**
User: registration, authentication, logout, update, delete.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 .env
-node_modules
\ No newline at end of file
+node_modules
+dist
```
Comment (esinitsa) [inline .gitignore]: Remove this folder from git

File: package.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
     "express-session": "1.16.2",
     "mongodb": "3.3.2",
     "mongoose": "5.7.1",
```
Comment (esinitsa) [inline package.json]: Do u really use this?

File: src/index.ts [hunk 3]
```diff
@@ -1,50 +1,36 @@
...
+app.get('/', posts.get);
+app.get('/posts/:id', currentPost.get);
+app.post('/post/create', createPost.post);
+app.post('/post/update', updatePost.post);
+app.post('/post/delete', deletePost.post);
 
+app.get('/users', users.get);
+app.post('/login', login.post);
+app.post('/logout', logout.post);
+app.post('/user/create', createUser.post);
```
Comment (esinitsa) [inline src/index.ts]: Use POST DELETE PUT GET types of requests for this

File: src/logic/post.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import express from "express";
```
Comment (esinitsa) [inline src/logic/post.ts]: Bad folder name

File: src/logic/post.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import models from "../models/index";
+
+export default class Post {
+
+  async showPosts(req: express.Request, res: express.Response) {
+    const result = await models.post.find();
+    res.send(result);
+  }
+
+  async showCurrentPost(req: express.Request, res: express.Response) {
```
Comment (esinitsa) [inline src/logic/post.ts]: `  async showCurrentPost({body : {id}}: express.Request, res: express.Response) {`

File: src/logic/post.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    const result = await models.post.findOne({ 'id': req.body.id });
+    res.send(result);
+  }
+
+  async createPost(req: express.Request, res: express.Response) {
+    const result = await models.post.create({
```
Comment (esinitsa) [inline src/logic/post.ts]: Add validations

File: src/logic/post.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    const result = await models.post.find();
+    res.send(result);
+  }
+
+  async showCurrentPost(req: express.Request, res: express.Response) {
+    const result = await models.post.findOne({ 'id': req.body.id });
+    res.send(result);
+  }
+
+  async createPost(req: express.Request, res: express.Response) {
```
Comment (esinitsa) [inline src/logic/post.ts]: Use object destruction

File: src/logic/post.ts [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+      text: req.body.text,
+      postedBy: req.body.postedBy
+    });
+    res.send(result);
+  }
+
+  async updatePost(req: express.Request, res: express.Response) {
+    const result = await models.post.findOne({ _id: req.body.id });
+    if (result) {
+      console.log(result);
```
Comment (esinitsa) [inline src/logic/post.ts]: Remove all unnecessary logs

File: src/logic/post.ts [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+      const currentUser = await models.user.findOne({ _id: result.postedBy });
+      if (currentUser) {
+        if (currentUser.isAuthenticated == true) {
```
Comment (esinitsa) [inline src/logic/post.ts]: ‘    if (currentUser.isAuthenticated) {’

File: src/logic/post.ts [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+          models.post.findByIdAndUpdate(result.id, {
+            text: req.body.text
+          }, { new: true }, (err, result) => {
+            if (err) return console.log(err);
```
Comment (esinitsa) [inline src/logic/post.ts]: Omg

File: src/logic/post.ts [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+          });
```
Comment (esinitsa) [inline src/logic/post.ts]: Omg

File: src/logic/post.ts [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+
+  async createPost(req: express.Request, res: express.Response) {
+    const result = await models.post.create({
+      text: req.body.text,
+      postedBy: req.body.postedBy
+    });
+    res.send(result);
+  }
+
+  async updatePost(req: express.Request, res: express.Response) {
```
Comment (esinitsa) [inline src/logic/post.ts]: Object destruction

File: src/logic/post.ts [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+        }
+      } else {
+        res.send("Issues with User");
+      }
+    } else {
+      res.send("Check id of the post");
+    }
+  }
+
+  async deletePost(req: express.Request, res: express.Response) {
```
Comment (esinitsa) [inline src/logic/post.ts]: Stupid logic

File: src/logic/post.ts [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+    const result = await models.post.findOne({ _id: req.body.id });
+    if (result) {
+      const currentUser = await models.user.findOne({ _id: result.postedBy });
+      if (currentUser) {
+        if (currentUser.isAuthenticated == true) {
+          models.post.findByIdAndDelete(result._id, (err, doc) => {
+            if (err) return console.log(err);
+          });
+          res.send("success!!");
+        } else {
```
Comment (esinitsa) [inline src/logic/post.ts]: Too many responses

File: src/logic/user.ts [hunk 15]
```diff
@@ -0,0 +1,94 @@
+import express from "express";
+import models from "../models/index";
+import Post from "../logic/post";
+
+const postMethods = new Post;
```
Comment (esinitsa) [inline src/logic/user.ts]: How it works?

File: src/logic/user.ts [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+
+export default class User {
+
```
Comment (esinitsa) [inline src/logic/user.ts]: Empty line

File: src/logic/user.ts [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+  async showUsers(req: express.Request, res: express.Response) {
+    const result = await models.user.find();
```
Comment (esinitsa) [inline src/logic/user.ts]: Do you really put passwords to response?

File: src/logic/user.ts [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+    res.send(result);
+  }
+
+  async createUser(req: express.Request, res: express.Response) {
```
Comment (esinitsa) [inline src/logic/user.ts]: Use object destruction

File: src/models/index.ts [hunk 19]
```diff
@@ -1,19 +1,19 @@
...
+import mongoose from "mongoose";
 
 import user from "./user";
 import post from "./post";
 
 const connectDB = () => {
     const { DATABASE_URL } = process.env;
-    if(!DATABASE_URL) {
+    if (!DATABASE_URL) {
         throw '123123123';
```
Comment (esinitsa) [inline src/models/index.ts]: Fuckup

File: src/logic/user.ts [hunk 20]
```diff
@@ -0,0 +1,94 @@
...
+      gender: req.body.gender,
+      isAuthenticated: false
+    });
+    res.send(result);
+  }
+  // CHEACK deleteAllPostOfUser
+  async deleteUser(req: express.Request, res: express.Response) {
+    const result = await models.user.findOne({ username: req.body.username });
+    if (result) {
+      if (result.isAuthenticated == true) {
```
Comment (MeyerPV) [inline src/logic/user.ts:27]: It's looks bad. Rewrite this. Avoid using a lot of **if/else** and ```especially``` nested **if/else**

**Pull request closed without merging**